### PR TITLE
fix: Update multi-region test in `avm/res/document-db/database-account`

### LINKED ISSUE #3588: [Failed pipeline] avm.res.document-db.database-account
Failed run: https://github.com/Azure/bicep-registry-modules/actions/runs/11390119682

### PR DESCRIPTION
**Pull request by seesharprun**
## Description

Updating the *multiRegion* test to solve the current pipeline failure.

- [x] Fixes #3588
- [x] Changed multi-region test to use hard-coded `eastus2` and `westus3`
- [x] Changed zone redundancy default to `false`

## Pipeline Reference

| Pipeline |
| -------- |
| [avm.res.document-db.database-account](https://github.com/seesharprun/bicep-registry-modules/actions/workflows/avm.res.document-db.database-account.yml) |

## Type of Change

- [ ] Update to CI Environment or utilities (Non-module affecting changes)
- [ ] Azure Verified Module updates:
  - [x] Bugfix containing backwards-compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`:
    - [x] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
    - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [ ] Feature update backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] Breaking changes and I have bumped the MAJOR version in `version.json`.
  - [ ] Update to documentation

## Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [ ] I have run `Set-AVMModule` locally to generate the supporting module files.
- [ ] My corresponding pipelines / checks run clean and green without any errors or warnings

### CODE REVIEW

**Comment by seesharprun** [comment]
This is NOT ready for review. I'm still running tests locally.

**Comment by seesharprun** [comment]
This can be reviewed, but let's hold off on merging this until we're ABSOLUTELY sure this will fix the tests in every environment.

**Comment by seesharprun** [comment]
@azure/avm-res-documentdb-databaseaccount-module-owners-bicep, take a look at this and advise on next steps.

While this works in my environment, I don't know how to validate that it will work in your environment.

File: avm/res/document-db/database-account/main.bicep
Comment (bryansan-msft) [inline avm/res/document-db/database-account/main.bicep:206]: Per AVM guidelines zone redundant needs to be always true

**Review by ghost — CHANGES_REQUESTED** [review]

File: avm/res/document-db/database-account/main.bicep
Comment (bryansan-msft) [reply avm/res/document-db/database-account/main.bicep:206]: Just FYI https://azure.github.io/Azure-Verified-Modules/faq/#what-does-avm-mean-by-waf-aligned

**Comment by bryansan-msft** [comment]
> @Azure/avm-res-documentdb-databaseaccount-module-owners-bicep, take a look at this and advise on next steps.
> 
> While this works in my environment, I don't know how to validate that it will work in your environment.



> @Azure/avm-res-documentdb-databaseaccount-module-owners-bicep, take a look at this and advise on next steps.
> 
> While this works in my environment, I don't know how to validate that it will work in your environment.

As long as it works in the main AVM pipelines I am sure it will work in my environment. As far as I understood from this PR you are just changing the main and paired region.

File: avm/res/document-db/database-account/main.bicep
Comment (seesharprun) [reply avm/res/document-db/database-account/main.bicep:206]: Unfortunately, the product team suggested against this. Capacity for zone-redundant accounts is constrained in some PAYG Azure subscriptions and other subscriptions we use for internal testing.

Some of the pipeline error messages directly call out the fact that the target subscription doesn't have capacity for a zonal account.

So, I'm leaning very heavily on the *[..], where possible and appropriate [...]* line from that WAF-aligned standard. I would argue here that zone redundancy may not be possible or appropriate as a default based on current capacity.

@markjbrown, can you provide some more detail here.

Comment (seesharprun) [reply avm/res/document-db/database-account/main.bicep:206]: Let's leave this review open while we discuss current product capacity and the WAF-aligned standard here: https://github.com/Azure/bicep-registry-modules/issues/3588#issuecomment-2428562666

Comment (seesharprun) [reply avm/res/document-db/database-account/main.bicep:206]: Per https://github.com/Azure/bicep-registry-modules/issues/3588#issuecomment-2429673345, they do want this change reverted. Unfortunately, this means that we will need to accept some possibility that the e2e and multi-region tests fail due to capacity.

I reverted this change and this should resolve the current conversation.

**Comment by eriqua** [comment]
@seesharprun what we could offer is to merge this contribution to a branch once ready for approval, so that we can test in the target AVM environment before it gets merged to main.

Note: if the failure caused a missing publishing, I also suggest to make a small update to the main.bicep, resulting in the update of the main.json as well. e.g. update a typo or a description.
That would qualify and trigger the CI to run the publishing step.

**Comment by seesharprun** [comment]
> [...] what we could offer is to merge this contribution to a branch once ready for approval, so that we can test in the target AVM environment before it gets merged to main.
> 
> Note: if the failure caused a missing publishing, I also suggest to make a small update to the main.bicep, resulting in the update of the main.json as well. e.g. update a typo or a description. That would qualify and trigger the CI to run the publishing step.

That would be amazing @eriqua! I just tested in my own subscription, and everything is green. If your team is willing to test this branch in your environment, that would help ensure that this PR "actually" fixes the underlying issue.

Thanks for the offer. This PR is ready for your test and the final review.

CC: @bryansan-msft

**Comment by eriqua** [comment]
> > [...] what we could offer is to merge this contribution to a branch once ready for approval, so that we can test in the target AVM environment before it gets merged to main.
> > Note: if the failure caused a missing publishing, I also suggest to make a small update to the main.bicep, resulting in the update of the main.json as well. e.g. update a typo or a description. That would qualify and trigger the CI to run the publishing step.
> 
> That would be amazing @eriqua! I just tested in my own subscription, and everything is green. If your team is willing to test this branch in your environment, that would help ensure that this PR "actually" fixes the underlying issue.
> 
> Thanks for the offer. This PR is ready for your test and the final review.
> 
> CC: @bryansan-msft

@seesharprun @bryansan-msft tested successfully in the upstream environment
[avm.res.document-db.database-account](https://github.com/Azure/bicep-registry-modules/actions/workflows/avm.res.document-db.database-account.yml)

**Review by ghost — APPROVED** [review]

**Pull request merged**
